mullvad-daemon/src/version/check.rs

```diff
@@ -409,25 +393,21 @@ fn version_check_inner(
         architecture,
         mullvad_update::version::IGNORE,
         min_metadata_version,
+        api.platform_version.clone(),
     );
     async move {
-        let (
-            current_version_supported,
-            AppVersionResponse2 {
-                version_info,
-                metadata_version,
-            },
-        ) = tokio::try_join!(supported_fut, v2_endpoint).map_err(Error::Download)?;
+        let v2_result = v2_endpoint.await.map_err(Error::Download)?;
 
         Ok(VersionCache {
-            current_version_supported,
-            version_info,
-            metadata_version,
+            current_version_supported: v2_result.current_version_supported,
+            version_info: v2_result.version_info,
+            #[cfg(in_app_upgrade)]
+            metadata_version: v2_result.metadata_version,
```

Comment:
This should be used on Linux now as well. Please look up all references to it.

Reply:
Done!